Life.java

```diff
@@ -42,11 +44,70 @@ public void setDead(int x, int y) {
   }
 
   public boolean isAlive(int x, int y) {
-    return raster[x][y];
+  // Verhindern von null Pointer Exception
+  if (x >= 0 && x < raster.length){
+    if (y >= 0 && y < raster[x].length){
+      return raster[x][y];
+    }
+  }
+    //Außerhalb des Rasters = tot
+    return false;
   }
 
   public ILife nextGeneration() {
-    // TODO Auto-generated method stub
-    return null;
+    Life l = new Life();
+    for (int i = 0; i< raster.length; i++){
+      for (int j = 0; j< raster[i].length; j++){
+        int neighbours = checkNeighbours(i,j);
+        if (neighbours < 2 || neighbours > 3){
```

Comment:
man könnte die einzelnen Regeln natürlich auch in eigene Methoden auslagern und entsprechend benennen.